RuntimeLibrary/plsan_shadow.cpp

```diff
@@ -0,0 +1,20 @@
+#include "plsan_shadow.h"
+
+#include <sys/mman.h>
+
+namespace __plsan {
+
+PlsanShadow::PlsanShadow() {
+  shadow_addr = mmap(NULL, MMAP_SIZE, PROT_READ | PROT_WRITE,
+                     MAP_ANONYMOUS | MAP_PRIVATE | MAP_NORESERVE, -1, 0);
+  if (shadow_addr == MAP_FAILED)
+    throw "mmap failed\n";
+}
+
```

Comment:
when program is over, we have to deallocate shadow memory.
So, add a Destructor. Like this
```
PlsanShadow::~PlsanShaow(){
    munmap(shadow_addr, MMAP_SIZE);
}
```